frontend/app-development/features/appPublish/components/DeployDropdown.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useState } from 'react';
+import classes from './DeployDropdown.module.css';
+import { AltinnConfirmDialog } from 'app-shared/components';
+import { StudioButton, StudioSpinner } from '@studio/components';
+import { Alert, Combobox, Spinner } from '@digdir/design-system-react';
+import type { ImageOption } from './ImageOption';
+import { useTranslation } from 'react-i18next';
+import { useAppReleasesQuery } from 'app-development/hooks/queries';
+import { BuildResult } from 'app-shared/types/Build';
+import { formatDateTime } from 'app-shared/pure/date-format';
+import { useStudioUrlParams } from 'app-shared/hooks/useStudioUrlParams';
+
+export interface DeployDropdownProps {
+  appDeployedVersion: string;
+  disabled: boolean;
+  setSelectedImageTag: (tag: string) => void;
+  selectedImageTag: string;
+  startDeploy: () => void;
+  isPending: boolean;
+}
+
+export const DeployDropdown = ({
+  appDeployedVersion,
+  selectedImageTag,
+  setSelectedImageTag,
+  disabled,
+  startDeploy,
+  isPending,
+}: DeployDropdownProps) => {
+  const { org, app } = useStudioUrlParams();
+  const [isConfirmDeployDialogOpen, setIsConfirmDeployDialogOpen] = useState<boolean>();
+  const { t } = useTranslation();
+
+  const {
+    data: releases = [],
+    isPending: releasesIsPending,
+    isError: releasesIsError,
+  } = useAppReleasesQuery(org, app, { hideDefaultError: true });
+
+  if (releasesIsPending)
+    return (
+      <StudioSpinner showSpinnerTitle={false} spinnerTitle={t('app_deployment.releases_loading')} />
+    );
+
+  if (releasesIsError) return <Alert severity='danger'>{t('app_deployment.releases_error')}</Alert>;
+
+  const imageOptions: ImageOption[] = releases
+    .filter((image) => image.build.result === BuildResult.succeeded)
+    .map((image) => ({
+      value: image.tagName,
+      label: `Version ${image.tagName} (${formatDateTime(image.created)})`,
```

Comment:
Could we move this text into the texts file to make it translatable?

Reply:
Done 👍